src/java.base/unix/classes/sun/nio/fs/UnixFileSystemProvider.java

```diff
@@ -350,12 +350,40 @@ public void checkAccess(Path obj, AccessMode... modes) throws IOException {
             mode |= X_OK;
         }
         try {
-            access(file, mode);
+            int errno = access(file, mode);
+            if (errno != 0)
+                throw new UnixException(errno);
         } catch (UnixException exc) {
             exc.rethrowAsIOException(file);
         }
     }
 
+    @Override
+    public boolean isReadable(Path path) {
+        UnixPath file = UnixPath.toUnixPath(path);
+        file.checkRead();
+        return access(file, R_OK) == 0 ? true : false;
```

Comment:
Just curious, why do we use redundant `? true : false` for a boolean value? Such a pattern did not exist in this file before this patch.

Reply:
> why do we use redundant `? true : false` for a boolean value?

@liach What meaning do you intend by "redundant" here?

Reply:
I mean we can just use `access(file, R_OK) == 0`

Reply:
One of the iterations did change the "exists" methods, the three isXXXX methods were missed. It's not a correctness issue of course, just a cleanup.

Reply:
Thanks. I intentionally did not integrate it yet so I will fix those first.